x/distribution/client/cli/tx.go

```diff
@@ -40,6 +45,28 @@ func GetTxCmd(storeKey string, cdc *amino.Codec) *cobra.Command {
 	return distTxCmd
 }
 
+func splitGenerateOrBroadcast(cliCtx context.CLIContext, txBldr authtxb.TxBuilder, msgs []sdk.Msg) error {
+	chunkSize := viper.GetInt(flagMaxMessagesPerTx)
```

Comment:
I'd pass this as an argument to make it easier to test the function (i.e. no need to set a variable via viper)

Reply:
We anyway need to passthrough the CLIContext and will force extracting the option in every single caller. 

This function is planned to eventually wrap/replace `utils.GenerateOrBroadcastMsgs` so we can have splitting available to most commands. Adding this as an argument would force extracting the value from viper in every single caller.

Reply:
We should either pass it as an argument or add it to the context now. Also, I don't see why we'd add it the context because I really only see this affecting this single command.

Reply:
Because right now this is affecting a single command as there is some urgency in getting this feature active, however, this is a problem that may affect any command.. 
In my opinion, the option to split should be available in all commands in the short term.
Also, as previously discussed, it is important that there is a discussion around ensuring future atomic transactions, etc. But as @zmanian and @alexanderbez commented, this does not seem to be a requirement at this moment.

Reply:
Until we decide on such a course of action, can we have it as a parameter for now? It should not be hard to change.